src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1278,9 +1278,30 @@ impl Coordinator {
         plan::DropObjectsPlan {
             drop_ids,
             object_type,
-            referenced_ids: _,
+            referenced_ids,
         }: plan::DropObjectsPlan,
     ) -> Result<ExecuteResponse, AdapterError> {
+        let referenced_ids_hashset = referenced_ids.iter().collect::<HashSet<_>>();
+
+        for obj_id in &drop_ids {
+            if !referenced_ids_hashset.contains(&obj_id) {
+                let notice_msg = match obj_id {
+                    ObjectId::Cluster(id) => format!("Cluster ID: {}", id),
+                    ObjectId::ClusterReplica((cluster_id, replica_id)) => {
+                        format!("Cluster Replica ID: ({}, {})", cluster_id, replica_id)
+                    }
+                    ObjectId::Database(id) => format!("Database ID: {}", id),
+                    ObjectId::Schema((db_specifier, schema_specifier)) => {
+                        format!("Schema ID: ({:?}, {:?})", db_specifier, schema_specifier)
+                    }
+                    ObjectId::Role(id) => format!("Role ID: {}", id),
+                    ObjectId::Item(id) => format!("Item ID: {}", id),
+                };
+
+                session.add_notice(AdapterNotice::CascadeDroppedObject { notice_msg });
```

Comment:
happy to change the `notice_msg` to your liking

Reply:
In general, notices are meant to be consumed by people and therefore should be as human readable as possible. I think we should try and mimic PostgreSQL's message as much as possible for this. If we take a look at an example from PostgreSQL it looks like this:


```
postgres=# CREATE TABLE t (a INT);
CREATE TABLE
postgres=# CREATE VIEW v1 AS SELECT * FROM t;
CREATE VIEW
postgres=# CREATE VIEW v2 AS SELECT * FROM t;
CREATE VIEW
postgres=# DROP TABLE t CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to view v1
drop cascades to view v2
DROP TABLE

```

If we look at the same example from this branch, it looks like this:

```
materialize=> CREATE TABLE t (a INT);
CREATE TABLE
materialize=>  CREATE VIEW v1 AS SELECT * FROM t;
CREATE VIEW
materialize=> CREATE VIEW v2 AS SELECT * FROM t;
CREATE VIEW
materialize=> DROP TABLE t CASCADE;
INFO:  Item ID: u2
INFO:  Item ID: u3
DROP TABLE

```

So instead of one notice per dropped object, we should have a single notice that contains all dropped objects.

The contents of the notice, which comes from here: https://github.com/MaterializeInc/materialize/blob/c244ed46a11b91953523343bbdd81d31358d904e/src/adapter/src/notice.rs#L267-L268

should look something like:
```Rust
write!(f, "drop cascades to {num_objects} other objects")
```

The contents of the detail, which comes from here: https://github.com/MaterializeInc/materialize/blob/c244ed46a11b91953523343bbdd81d31358d904e/src/adapter/src/notice.rs#L185-L186

should look something like:
```Rust
Some(objects.iter()
  .map(|object| format!("drop cascades to {} {}", object.type(), object.name())
  .join("\n"))
```

That's psuedo-code, there's no `object.type()` or `object.name()` functions. Instead, you'll have to look that up in the catalog via the ID. For example clusters will probably look like this:

```Rust
let typ = ObjectType::Cluster.to_string();
let name = self.catalog().get_cluster(*id).name().to_string();
```

Items will probably looks like this:

```Rust
let entry = self.catalog().get_entry(id);
let typ = entry.item_type().to_string();
let name = self.catalog().resolve_full_name(entry.name(), Some(session.conn_id())).to_string();
```

If you get stuck on any of the other object types let me know and I'll provide some help.